### ISSUE TITLE
[MMAI] Pending fixes and improvements

### ISSUE THREAD

**Issue by smanolloff** [vcmi/vcmi#6618]
This issue serves as an (evolving) task-list of MMAI fixes and improvements planned for the (near) future. Once actual work begins on a task, a separate issue for it can be opened (not sure if needed).

- [x] Mark RETREAT action returned by NNModel as invalid
  Discovered in #6613. The model should never return this action. Marking it invalid would retry the inner getAction loop, preventing makeAction packet round-trip and server error messages. Possibly caused by corrupted state sent to the model, so rebuilding the `State` object before retrying the loop might help.
  * Implemented in https://github.com/vcmi/vcmi/pull/6665 
  * Fix for the likely root cause: https://github.com/vcmi/vcmi/pull/6664 

- [x] NNModel should return action probabilities to allow client-side sampling for a more interesting behaviour. Sampling is still deterministic, controlled via seed, but unpredictable by a human. Currently it is impossible as the model returns logits conditioned on already sampled act0 and hex1. ~Ideally, a single probability distribution over N_ACTIONS should be returned instead~.
  * Implemented in https://github.com/vcmi/vcmi/pull/6679

- [x] Bump value of cloned units. Currently they appear as regular summons which have their value _greatly decreased_, hence the model mostly ignores them. Instead, clones should have a _greatly increased_ value to make them preferred targets.
  * Implemented in https://github.com/vcmi/vcmi/pull/6666

- [x] Fix bug with incorrectly calculated action mask [here](https://github.com/smanolloff/vcmi/blob/9d6ec17f48bd141b1dda669127a39c38114e693e/AI/MMAI/BAI/v13/hex.cpp#L357) (should be `continue` instead of `return`). This means the models were prevented from attacking in some cases where a friendly creature occupies a nearby hex. Removing this limitation should make MMAI stronger (may require a new training session).
  - Implemented in https://github.com/vcmi/vcmi/pull/6680

- [x] Release `mmai` mod version 1.3 which contains 3 types of models: static, dynamic and stochastic-sampling enabled models (so it works with VCMI `1.7.0`..`1.7.2`). The mod must have max supported VCMI version 1.7.9, as there will be MMAI refactoring in 1.8.0 (see corresponding list item for refactoring)
  - mmai mod version 1.3 [released](https://github.com/vcmi-mods/mmai/commit/eceec1f15647160d86aef3f555527dd5d40a6036) with supported vcmi versions `1.7.0`..`1.7.9`. Tested on android with 1.7.1 - works ok (uses non-stochastic models). Once 1.7.2 is out stochastic models should be picked up automatically (mod contains both)
  - mmai mod version 2.0.0 [released](https://github.com/vcmi-mods/mmai/commit/711a5664dd96eebe26cc67bee02877b3d7609b68) with supported vcmi versions `1.8.0`..`1.8.9` (current `develop`). Includes only stochastic models.

- [x] Refactor MMAI and cleanup backwards-compatible changes from 1.7.X hotfixes, dropping support for several iterations of old (non-dynamic, non-stochastic) models. This should target VCMI 1.8
  - Implemented in https://github.com/vcmi/vcmi/pull/6687

- [x] Create siege-only model with a more defensively-oriented playstyle. This will allow to re-release the improved VIP-trained model from [this](https://github.com/vcmi-mods/mmai/commit/78ed5a6a168c0735ff6b0f96c83c598da5104fff) release
  - Implemented in https://github.com/vcmi/vcmi/pull/7137

- [ ] Tactics support (scripted)

**Comment by IvanSavenko** [comment]
Closing, all items implemented
